nano/core_test/backlog.cpp

```diff
@@ -75,32 +75,44 @@ TEST (backlog, population)
 	nano::test::system system{};
 	auto & node = *system.add_node ();
 
-	nano::mutex mutex;
-	std::unordered_set<nano::account> activated;
-
-	node.backlog.activate_callback.add ([&] (nano::transaction const & transaction, nano::account const & account, nano::account_info const & account_info, nano::confirmation_height_info const & conf_info) {
-		nano::lock_guard<nano::mutex> lock{ mutex };
-
-		activated.insert (account);
+	// the node destruction happens in the background and parts of the node object
```

Comment:
Well, this comment is not true. `system` destructor ensures node will be stopped and therefore all relevant threads will also be stopped.

Reply:
I am certain that parts of the node survive even after system destruction. I have spent a month last year due to that problem and in the end I resorted to this pattern with weak pointers.

Reply:
You are confusing IO threads with component threads.

Reply:
Resorting to weak_ptr is a sign of not having a well managed lifetime of components and usually not necessary. It's possible to have that for IO threads too, the PR for that was mean to land after V24 release.